tests/stack/test_api.py

```diff
@@ -0,0 +1,258 @@
+        """
+            _load_json should read the same JSON content that was originally in the
+            stack configuration JSON.
+        """
+        stack_path = os.path.join(tmpdir.strpath, TEST_STACK_PATH)
+        os.makedirs(os.path.dirname(stack_path))
+        with open(stack_path, "w+") as f:
+            json.dump(TEST_STACK, f)
+        config = stack_api._load_json(stack_path)
+        assert config == TEST_STACK
+
+    def test_generate_stack_status_path(self, stack_api, tmpdir):
+        """
+            The _generate_stack_status_path should add the word 'deployed' between the json file
+            extension and the filename of the stack configuration file.
+        """
+        config_path = os.path.join(tmpdir.strpath, 'test.json')
+        expected_status_path = os.path.join(tmpdir.strpath, 'test.deployed.json')
+        generated_path = stack_api._generate_stack_status_path(config_path)
+        assert expected_status_path == generated_path
+
+        config_path = os.path.join(tmpdir.strpath, 'test.st-ack.json')
+        expected_status_path = os.path.join(tmpdir.strpath, 'test.st-ack.deployed.json')
+        generated_path = stack_api._generate_stack_status_path(config_path)
+        assert expected_status_path == generated_path
+
+    def test_save_load_stack_status(self, stack_api, tmpdir):
+        """
+            When saving the a stack status through _save_stack_status, it should be able to be
+            loaded by _load_stack_status and have the same exact contents.
+        """
+        config_path = os.path.join(tmpdir.strpath, 'test.json')
+        status_path = stack_api._generate_stack_status_path(config_path)
+        stack_api._save_json(status_path, TEST_STATUS)
+
+        status = stack_api._load_json(status_path)
+        assert status == TEST_STATUS
+
+    def test_deploy_relative_paths(self, stack_api, tmpdir):
+        """
+            When doing stack_api.deploy, in every call to stack_api._deploy_resource, the current
+            working directory should be the same directory as where the stack config template is
+            contained so that relative paths for resources can be relative to the stack config
+            instead of where CLI calls the API functions.
+        """
+        config_working_dir = os.path.join(tmpdir.strpath, 'stack')
+        config_path = os.path.join(config_working_dir, 'test.json')
+        os.makedirs(config_working_dir)
+        with open(config_path, 'w+') as f:
+            json.dump(TEST_STACK, f)
+        initial_cwd = os.getcwd()
+
+        def _deploy_resource(resource, stack_status):
+            assert os.getcwd() == config_working_dir
+            return TEST_JOB_STATUS
+
+        stack_api._deploy_resource = mock.Mock(wraps=_deploy_resource)
+        stack_api.deploy(config_path)
+        assert os.getcwd() == initial_cwd  # Make sure current working directory didn't change
+
+    def test_deploy_error_path(self, stack_api, tmpdir):
```

Comment:
I don't think you need this test since the working directory should be a property of the current process.